ui/app/pages/confirm-token-transaction-base/confirm-token-transaction-base.component.js

```diff
@@ -11,30 +13,36 @@ import {
 import { getWeiHexFromDecimalValue } from '../../helpers/utils/conversions.util'
 import { ETH, PRIMARY } from '../../helpers/constants/common'
 
-export default class ConfirmTokenTransactionBase extends Component {
-  static contextTypes = {
-    t: PropTypes.func,
-  }
+const ConfirmTokenTransactionBase = ({
+  toAddress,
+  tokenAddress,
+  tokenAmount = '0',
+  tokenSymbol,
+  fiatTransactionTotal,
+  ethTransactionTotal,
+  contractExchangeRate,
+  conversionRate,
+  currentCurrency,
+}) => {
 
-  static propTypes = {
-    tokenAddress: PropTypes.string,
-    toAddress: PropTypes.string,
-    tokenAmount: PropTypes.string,
-    tokenSymbol: PropTypes.string,
-    fiatTransactionTotal: PropTypes.string,
-    ethTransactionTotal: PropTypes.string,
-    contractExchangeRate: PropTypes.number,
-    conversionRate: PropTypes.number,
-    currentCurrency: PropTypes.string,
-  }
+  const t = useContext(I18nContext)
+  const hexWeiValue = useMemo(() => {
+    if (tokenAmount === '0' || !contractExchangeRate) {
+      return '0'
+    }
 
-  static defaultProps = {
-    tokenAmount: '0',
-  }
+    const decimalEthValue = (
+      (new BigNumber(tokenAmount)).mul(new BigNumber(contractExchangeRate))
+    ).toFixed()
 
-  getFiatTransactionAmount () {
-    const { tokenAmount, currentCurrency, conversionRate, contractExchangeRate } = this.props
+    return getWeiHexFromDecimalValue({
+      value: decimalEthValue,
+      fromCurrency: ETH,
+      fromDenomination: ETH,
+    })
+  }, [tokenAmount, contractExchangeRate])
 
+  const getFiatTransactionAmount = () => {
```

Comment:
These functions are now getting declared and executed upon each render, so it's now slightly worse performing than it was before, and not particularly idiomatic.

You could instead move these functions to module scope, and pass in the props required. Or you could store these values/sub-components in local variables, rather than using functions.

Reply:
Good point, I'll do that.